dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/MetricsFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.metrics;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.extension.Adaptive;
+import org.apache.dubbo.common.extension.ExtensionScope;
+import org.apache.dubbo.common.extension.SPI;
+
+/**
+ * MetricsFactory. (SPI, Singleton, ThreadSafe)
+ */
+@SPI(value = "prometheus", scope = ExtensionScope.APPLICATION)
+public interface MetricsFactory {
+
+    /**
+     * Create metrics.
+     *
+     * @param url
+     * @return metrics
+     */
+    @Adaptive("protocol")
+    Metrics getMetrics(URL url);
```

Comment:
The name of `Metrics` / `MetricsFactory` is not very appropriate, here is a suggestion: `MetricsProvider` / `MetricsProviderFactory`

Reply:
MetricsProvider also seems inappropriate. What capabilities can Prometheus' SPI provide? Think about this interface

Reply:
Currently deleted. Wait for next pull request to complete the implementation